File: App_Vapor/Server/DataTransferObjects/UserForRegistrationDto.cs
using System.ComponentModel.DataAnnotations;

namespace Server.DataTransferObjects
{
    public class UserForRegistrationDto
    {
        [Required(ErrorMessage = "FullName is required.")]
        public string? FullName { get; set; }
        [Required(ErrorMessage = "Email is required.")]
        public string? Email { get; set; }
        [Required(ErrorMessage = "Password is required.")]
        public string? Password { get; set; }
        [Compare(nameof(Password), ErrorMessage = "The password and confirmation password do not match.")]
        public string? ConfirmPassword { get; set; }
    }

    public class RegistrationResponseDto
    {
        public bool IsSuccessfulRegistration { get; set; }
        public IEnumerable<string>? Errors { get; set; }
    }
}

File: Front-end/client/src/app/interfaces/user-for-registration-dto.ts
export interface UserForRegistrationDto {
  fullName: string;
  email: string;
  password: string;
  confirmPassword: string;
}

export interface RegistrationResponse {
  isSuccessfulRegistration: boolean;
  errors: string[];
}

// interfaz para el login
export interface UserForAuthenticationDto {
  email: string;
  password: string;
}

// interfaz para la respuesta del login
export interface AuthResponseDto {
  isAuthSuccessful: boolean;
  errorMessage: string;
  token: string;
}